source/StellarObject.cpp

```diff
@@ -57,9 +57,10 @@ double StellarObject::Radius() const
 // If it is possible to land on this planet, this returns the Planet
 // objects that gives more information about it. Otherwise, this
 // function will just return nullptr.
+// Planets without sprites are not landable
 const Planet *StellarObject::GetPlanet() const
 {
-	return planet;
+	return HasSprite() ? planet : nullptr;
```

Comment:
Revert this.

If a ship is trying to land on a stellar object, that process should check for a sprite.

This method is a simple getter, and with this change _all_ of the code doesnt know that a spriteless StellarObject is attached to a planet.

Reply:
The comments above this function do specifically note that the function returns a planet *only* if it is landable, else it returns nullptr, so the overall logic is unchanged, but I see your point about cluttering up a simple getter.

The alternative is a lot more checks in various other places, from what I can see, logic in player ship and AI ship logic for running away, refuelling etc, I tried in the Planet::canland() function, but I couldnt see a way to get the body.hassprite() from Planet class.